src/Raven.Server/Documents/Operations/Operations.cs

```diff
@@ -187,16 +193,24 @@ void ProgressNotification(IOperationProgress progress)
                 RaiseNotifications(notification, operation);
             });
 
-            _active.TryAdd(id, operation);
+            Monitor.Enter(locker);
```

Comment:
it might hang forever if you will enter that from the Task first?

Maybe a better option would be to create a task (without starting it). Add it here and then start it?

Reply:
If we exit the monitor properly, we will hold this only very shortly, no?

Reply:
I changed the code to take the lock before running the operation task, so it cannot enter from the task first.